### PR TITLE
Create Sell Order and Search (Order Discovery)

### PR DESCRIPTION
**Pull request by th3o6a1d**

### CODE REVIEW

File: src/models/commons.ts [hunk 1]
```diff
@@ -12,11 +12,11 @@ export interface AssetType {
     * For example, for an asset class of ERC-20, the data holds the address of the token.
     * For ERC-721 data holds smart contract address and tokenId.
     **/
-    data: any,
+    data: any
 }
 
 export interface Asset {
-    amount: number,
+    value: number,
```
Comment (benefacto) [inline src/models/commons.ts:19]: I noticed you changed this. The docs in the contract say it is [an amount](https://github.com/rariblecom/protocol-contracts/blob/master/asset/contracts/LibAsset.md) which, I believe, is more informative than just using `value`. Do you disagree?

**Review by benefacto — CHANGES_REQUESTED** [review]
This is really great work! Especially given how little direction you had on this and that is is one of your early dOrg projects. I added some feedback, feel free to address if you think it merits it.

File: src/models/orders.ts [hunk 2]
```diff
@@ -1,40 +1,86 @@
...
+    token?: string,
+    tokenId?: BigInteger,
+    collection?: string
+}
+
+/**
+ * Sell Orders
+ **/
+
+export interface SellOrder extends Order {
```
[the remark this answers is not in the document]
Comment (th3o6a1d) [reply src/models/orders.ts:54]: I wondered that, and was definitely not trying to create new data structures unnecessarily, but when you look at the [response object of the API](https://api-reference.rarible.com/#operation/createOrUpdateOrder), there are a number of fields that are not included in the Order type. We can extend just by creating a bunch of optional fields I suppose.

File: src/models/commons.ts (hunk 1 above)
Comment (th3o6a1d) [reply src/models/commons.ts:19]: If you read the language closely, the field is titled `value` I believe.  In their other API examples, they use `value`. [Here](https://docs.rarible.com/exchange/creating-a-sell-order) for example

File: src/index.ts [hunk 3]
```diff
@@ -88,4 +97,13 @@ export declare class RaribleSDK {
    public getBuyOrder(
     filter: OrderFilter
   ): Promise<Order>;
+
+  /**
+   * Search orders
+   *
+   * @param {SearchFilter} filter - Defines criteria to filter orders by.
+   */
+   public searchOrders(
```
[the remark this answers is not in the document]
Comment (th3o6a1d) [reply src/index.ts:106]: Good call, yes, I noticed the API was down too.

File: src/models/orders.ts [hunk 4]
```diff
@@ -1,40 +1,86 @@
-import { Asset } from "./commons";
```
Comment (carlosfebres) [inline src/models/orders.ts:1]: We need to format the file using Prettier to not have avoidable merge-conflicts

**Review by carlosfebres — CHANGES_REQUESTED** [review]

**Pull request merged**